### PR TITLE
Improve readme + cleanup + test coverage 100% + psalm level 1

### PR DESCRIPTION
**Pull request by vjik**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ❌
| Fixed issues  | -

And throw exception on call `getCacheKey()` in counter without id.

### CODE REVIEW

File: tests/CounterTest.php [hunk 1]
```diff
@@ -43,10 +44,16 @@ public function testStatisticsShouldBeCorrectWhenLimitIsReached(): void
 
     public function testShouldNotBeAbleToSetInvalidId(): void
     {
-        $this->expectException(\LogicException::class);
+        $this->expectException(LogicException::class);
         (new Counter(10, 60, new ArrayCache()))->incrementAndGetState();
     }
 
+    public function testCacheKeyShouldNotBeAbleWithoutId(): void
```
Comment (samdark) [inline tests/CounterTest.php]:
```suggestion
    public function testGetCacheKeyShouldFailWithoutId(): void
```

File: src/Counter.php [hunk 2]
```diff
@@ -89,6 +82,10 @@ public function setTtlInSeconds(int $secondsTTL): void
      */
     public function getCacheKey(): string
     {
+        if ($this->id === null) {
+            throw new LogicException('The counter ID should be set.');
```
Comment (samdark) [inline src/Counter.php:86]: I'm not sure it should be required.

File: src/Counter.php [hunk 3]
```diff
@@ -41,16 +41,9 @@ final class Counter implements CounterInterface
 
     private int $ttlInSeconds = self::DEFAULT_TTL;
 
-    /**
-     * @var int Last increment time.
-     * In GCRA it's known as arrival time.
```
Comment (samdark) [inline src/Counter.php:46]: This info was kinda important.

File: src/Counter.php [hunk 4]
```diff
@@ -41,16 +41,9 @@ final class Counter implements CounterInterface
...
-     */
-    private int $lastIncrementTimeInMilliseconds;
```
Comment (samdark) [inline src/Counter.php:48]: @romkatsu do you remember why it was a property?

File: src/Counter.php (hunk 3 above)
Comment (vjik) [reply src/Counter.php:46]: Add comment to line 100, where created and used this variable.

File: src/Counter.php (hunk 2 above)
Comment (vjik) [reply src/Counter.php:86]: `getCacheKey()` can be call when id is `null`.

Comment (samdark) [reply src/Counter.php:86]: Yes and that could be alright.

File: src/Counter.php (hunk 4 above)
Comment (romkatsu) [reply src/Counter.php:48]: I do not remember. it seems that this is done to simplify the methods.

File: src/Counter.php (hunk 2 above)
Comment (samdark) [reply src/Counter.php:86]: OK. Re-checked. This is good.

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
Thanks!

**Pull request merged**